.github/workflows/functional-test.yml

```diff
@@ -22,4 +22,40 @@ jobs:
         run: |
           chmod +x run.sh
           bash run.sh
-        working-directory: v2/cmd/functional-test
\ No newline at end of file
+        working-directory: v2/cmd/functional-test
+  
+  functional-mac:
+    name: Functional Test - macos
+    runs-on: macos-latest
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
+
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Functional Tests
+        run: |
+          chmod +x run.sh
+          bash run.sh
+        working-directory: v2/cmd/functional-test
+
+  functional-windows:
+    name: Functional Test - windows
+    runs-on: windows-latest
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
+
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Functional Tests
+        run: |
+          chmod +x run.sh
+          bash run.sh windows
```

Comment:
With a matrix strategy and something like this I think it's possible to handle all cases with a unique job:
```
- name:  Functional Tests
  run:   |
         if [ "$RUNNER_OS" == "Windows" ]; then
              bash run.sh windows
         else
              bash run.sh
         fi
  ```

Reply:
Done